### ISSUE TITLE
How many training steps will get better results?

### ISSUE THREAD

**Issue by puckikk1202** [tencent-ailab/IP-Adapter#114]
Hi, thanks for your great work!
I currently training the ip_adaptor_plus model using a customized DeepFashion Mutli-Modal dataset (with image-text paired data). I have trained around 10k steps in s BS. (about 1 epoch of the dataset).
![image](https://github.com/tencent-ailab/IP-Adapter/assets/52399960/19fc58dd-1d46-4786-bbdd-e7e2c9a44041)
But MSE loss was not much dropping after around 2k steps as shown.
And I would like to ask two questions:
1. How many training steps will get a better result?
2. As shown in the image, there were some sampling results that showed up in multi-persons. Should I apply some neg_prompt (for example multi-persons in this case) to guide the training?
BTW, the sampling parameters followed the demo that you provided.
Looking forward to receiving your kind reply!

**Comment by xiaohu2015** [comment]
@puckikk1202 hi,  IP-Adapter is trained on a single machine with 8 V100 GPUs for 1M steps with a batch size of 8 per GPU. If you use a small dataset, I think you can finetune our pretrained ip-adapter.

**Comment by gagbaghdas** [comment]
@puckikk1202 Hi . Have you managed to train it?

**Comment by puckikk1202** [comment]
Yes, as mentioned above, the finetuned result is better than the train-from-scratch result.

**Comment by kadirnar** [comment]
> Yes, as mentioned above, the finetuned result is better than the train-from-scratch result.

Are you continuing to develop? I am also testing the VITON-HD dataset.

**Comment by puckikk1202** [comment]
> 



> > Yes, as mentioned above, the finetuned result is better than the train-from-scratch result.
> 
> Are you continuing to develop? I am also testing the VITON-HD dataset.

Hi, I am continuously testing the results on different pipelines. To ensure the quality, I recommend you to use the AnimateAnyone-liked reference net to inject the style image into SD unet. But for a plug-and-play lighted module, you can fine-tune ipadaptor as mentioned.

**Comment by kadirnar** [comment]
@puckikk1202 ,
I downloaded the VITON-HD images. I produce subtitles with Moondream2. Later I will train it with ip-adapter-plus. What do you recommend regarding the parameters?

I also recommend looking at this new study. The train code has not been published yet.

https://github.com/ShineChen1024/MagicClothing
